### PR TITLE
Added toggle switch for the feature of pokemon size by rarity

### PR DESCRIPTION
**Pull request by jagauthier**
Some people prefer their rare pokemon to be enlarged, and some do not.
This toggle, under "Style Settings" will allow it to be customized.

## How Has This Been Tested?
I tested this on my development map.

## Screenshots (if appropriate):
image
image
image
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: static/js/map.common.js [hunk 1]
```diff
@@ -1070,7 +1074,7 @@ function getGoogleSprite(index, sprite, displayHeight) {
     }
 }
 
-function setupPokemonMarkerDetails(item, map, scaleByRarity = true) {
+function setupPokemonMarkerDetails(item, map) {
```
Comment (pogo-excalibur) [inline static/js/map.common.js]: The `scaleByRarity` parameter is used by the stats page, which I don't see any edits for.

Comment (jagauthier) [inline static/js/map.common.js]: Huh. I did a find through the code and did not see that.  Let me check it out.  I can quite easily set it back.

Comment (jagauthier) [inline static/js/map.common.js]: ~/RocketMap$ find . -name "*" -exec grep "scaleByRarity" {} \; -print 2> /dev/null
~/RocketMap$

I'm not seeing anything.

Comment (BrownSlaughter) [inline static/js/map.common.js]: the full stats page when you click show all locations on a pokemon

File: static/js/map.common.js [hunk 2]
```diff
@@ -942,6 +942,10 @@ var StoreOptions = {
         default: false,
         type: StoreTypes.Boolean
     },
+    'sizeUp': {
```
Comment (BrownSlaughter) [inline static/js/map.common.js]: use something more suitable like upscale

File: static/js/map.common.js (hunk 1 above)
Comment (BrownSlaughter) [inline static/js/map.common.js]: as mentioned this was a fix for not upscaling pokemon on the full stats page

File: static/js/map.common.js [hunk 3]
```diff
@@ -1080,7 +1084,7 @@ function setupPokemonMarkerDetails(item, map, scaleByRarity = true) {
 
     var iconSize = (map.getZoom() - 3) * (map.getZoom() - 3) * 0.2 + Store.get('iconSizeModifier')
 
-    if (scaleByRarity) {
+    if (Store.get('sizeUp')) {
```
Comment (BrownSlaughter) [inline static/js/map.common.js]: This removes the previous fix

File: static/js/map.common.js [hunk 4]
```diff
@@ -1107,9 +1111,9 @@ function setupPokemonMarkerDetails(item, map, scaleByRarity = true) {
     return markerDetails
 }
 
-function setupPokemonMarker(item, map, isBounceDisabled, scaleByRarity = true) {
+function setupPokemonMarker(item, map, isBounceDisabled) {
```
Comment (BrownSlaughter) [inline static/js/map.common.js]: removing this " scaleByRarity = true" will upscale pokemon on the Full stats page

File: static/js/map.common.js [hunk 5]
```diff
@@ -1107,9 +1111,9 @@ function setupPokemonMarkerDetails(item, map, scaleByRarity = true) {
...
     // Scale icon size up with the map exponentially, also size with rarity.
-    const markerDetails = setupPokemonMarkerDetails(item, map, scaleByRarity)
+    const markerDetails = setupPokemonMarkerDetails(item, map)
```
Comment (BrownSlaughter) [inline static/js/map.common.js]: it was there for a reason for the previously mentioned fix

File: static/js/map.common.js [hunk 6]
```diff
@@ -1125,9 +1129,9 @@ function setupPokemonMarker(item, map, isBounceDisabled, scaleByRarity = true) {
     return marker
 }
 
-function updatePokemonMarker(item, map, scaleByRarity = true) {
+function updatePokemonMarker(item, map) {
```
Comment (BrownSlaughter) [inline static/js/map.common.js]: it was there for a reason for the previously mentioned fix

File: static/js/map.common.js [hunk 7]
```diff
@@ -1125,9 +1129,9 @@ function setupPokemonMarker(item, map, isBounceDisabled, scaleByRarity = true) {
...
     // Scale icon size up with the map exponentially, also size with rarity.
-    const markerDetails = setupPokemonMarkerDetails(item, map, scaleByRarity)
+    const markerDetails = setupPokemonMarkerDetails(item, map)
```
Comment (BrownSlaughter) [inline static/js/map.common.js]: it was there for a reason for the previously mentioned fix

File: static/js/map.js [hunk 8]
```diff
@@ -415,6 +415,7 @@ function initSidebar() {
     $('#sound-switch').prop('checked', Store.get('playSound'))
     $('#pokemoncries').toggle(Store.get('playSound'))
     $('#cries-switch').prop('checked', Store.get('playCries'))
+    $('#size-switch').prop('checked', Store.get('sizeUp'))
```
Comment (BrownSlaughter) [inline static/js/map.js]: Upscale-switch is more appropriate

File: static/js/map.js [hunk 9]
```diff
@@ -2730,6 +2731,9 @@ $(function () {
         Store.set('playCries', this.checked)
     })
 
+    $('#size-switch').change(function () {
```
Comment (BrownSlaughter) [inline static/js/map.js]: Upscale-switch is more appropriate

File: static/js/map.common.js (hunk 1 above)
Comment (jagauthier) [inline static/js/map.common.js]: Okay, thanks! I see now.

File: static/js/map.js [hunk 10]
```diff
@@ -2730,6 +2731,9 @@ $(function () {
...
+        Store.set('sizeUp', this.checked)
```
Comment (BrownSlaughter) [inline static/js/map.js]: Upscale is more appropriate than "sizeup"

File: templates/map.html [hunk 11]
```diff
@@ -379,6 +379,17 @@
           <h3><i class="fa fa-map-o fa-fw"></i>Style Settings</h3>
           <div>
             <div class="form-control switch-container">
+              <h3>Size Pokémon by rarity</h3>
```
Comment (BrownSlaughter) [inline templates/map.html]: Upscale pokemon by rarity

File: templates/map.html [hunk 12]
```diff
@@ -379,6 +379,17 @@
...
+              <div class="onoffswitch">
+                <input id="size-switch" type="checkbox" name="size-switch" class="onoffswitch-checkbox" checked>
+                <label class="onoffswitch-label" for="size-switch">
```
Comment (BrownSlaughter) [inline templates/map.html]: Upscale-switch is more appropriate

**Comment by BrownSlaughter** [comment]
if you are re working the scaleByRarity fix be sure to check how it works in statistics.js

**Comment by jagauthier** [comment]
The scaling on the statistics page worked fine for me.  I am not explicitly setting scaleByRarity to true, though.  Other tests are welcome @pogo-excalibur  (since you implemented the first fix)

**Comment by pogo-excalibur** [comment]
@jagauthier stats working fine again - 👍

**Comment by sLoPPydrive** [comment]
One thing is missing. It seems you have to reload the page when flipping the switch. This can easily be fixed by the following code in the switch change handler:
```javascript
// Change and store the flag
Store.set('upScale', this.checked)

// Remove all Pokemon markers from map
var oldPokeMarkers = []
$.each(mapData['pokemons'], function (key, pkm) {
    // for any marker you're turning off, you'll want to wipe off the range
    if (pkm.marker.rangeCircle) {
        pkm.marker.rangeCircle.setMap(null)
        delete pkm.marker.rangeCircle
    }
    pkm.marker.setMap(null)
    oldPokeMarkers.push(pkm.marker)
})
markerCluster.removeMarkers(oldPokeMarkers)
mapData['pokemons'] = {}

// Reload all Pokemon
lastpokemon = false
updateMap()
```

**Comment by jagauthier** [comment]
I appreciate that.  I did take a shot at it, but I really wasn't sure how it worked.  New pokemon take effect when they spawn, and simply zooming also worked.  I will implement your suggestion.

**Pull request closed without merging**